csv23/shortcuts.py

```diff
@@ -206,7 +206,7 @@ def write_csv(file, rows, header=None, dialect=DIALECT, encoding=ENCODING,
                 buf = f.buffer
                 for rows in iterslices(rows, 1000):
                     writer.writerows(rows)
-                    hashsum.update(buf.getbuffer())
+                    hashsum.update(bytes(buf.getbuffer()))
```

Comment:
I would have expected to find some performance degradation from copying the buffer instead of passing it, but I could not find it (at least from a quick check).

IIUC this is essentially a workaround. In that case I think I would prefer [something like this](https://github.com/xflr6/csv23/commit/c6330b86d3ade8ca8bfbc9a8d645bdd2079934fc#diff-6fc6dcdbbdc2e5344e59faf119962aa4), can you adapt?

Reply:
I don't think an actual copy will happen on any modern system. Even if Python doesn't elide it by taking the reference to the original buffer, system's CoW will elide copying until it is modified (i.e. it will never happen).

I'm fine with either solution. To be honest, I don't think the changes here can affect performance in any measurable way. However, in my opinion adding the conditional path as you suggest may have more overhead than calling `getvalue()` unconditionally.

Are you asking me to copy your version to mine? If you're just acking to my ack, then feel free to merge your version into mine and push it as is. My primary concern is getting the bug fixed, not my name in commit history ;-).